pkg/lightning/backend/tidb.go

```diff
@@ -87,7 +87,10 @@ func NewTiDBBackend(db *sql.DB, onDuplicate string) Backend {
 
 func (row tidbRow) ClassifyAndAppend(data *Rows, checksum *verification.KVChecksum, _ *Rows, _ *verification.KVChecksum) {
 	rows := (*data).(tidbRows)
-	*data = tidbRows(append(rows, row))
+	// I'm not sure if `rows := data.(*tidbRows); *rows = append(*rows, row)` could solve this lint.
+	// Seems there are some stories. Leave it untouched.
+	//nolint:gocritic
+	*data = append(rows, row)
```

Comment:
I think we can make the two slices as return parameters to avoid this lint but I'm not sure if it's necessary.

Reply:
I guess TiDB requires the `Row` interface too(due to log restore and BR in SQL). If that is real, I think when we want to refactor that, we'd better to make another PR.